### PR TITLE
Add holiday extensions to F1155, F1255

### LINKED ISSUE #175: Additional undocumented registers for F1255
I've moved from nibepi to nibegw via esphome in my HA installation, and I realized afterwards that I added additional undocumented registers in my old setup.

What is the policy for having these undocumented registers added to this library?

The registers I found by using my config backup are as follows:
Register | Factor | Size | Mode | Min | Max | Title | Info
----|----|----|----|----|----|----|----
48044 | 1 | u16 | Read-only? | 0 | 65535 | Holiday - Start date | Days since 2007-01-01
48045 | 1 | u16 | Read-only? | 0 | 65535 | Holiday - End date | Days since 2007-01-01
47052 | 1 | u32 | R/W | 0 | 86400 | Periodic HW time of day | Time of day in seconds when periodic HW is activated

For the periodic HW time register, I've had mixed results when writing to it. I believe it only allows writing whole hours as value.
And sadly I wasn't able to write the holiday dates to my F1255 using nibepi, which would've been useful.

### PR DESCRIPTION
**Pull request by elupus**
Fixes #175
Related to #174 

Verified that the field value of 10 is invalid on F1155, so it likely is the same problem on the F750.

### CODE REVIEW

**Comment by elupus** [comment]
What do you think, should we just add this to all F-series?

File: nibe/data/f1155_f1255.json [hunk 1]
```diff
@@ -5395,16 +5395,17 @@
...
     "min": 0.0,
-    "max": 10.0,
+    "max": 1.0,
     "default": 0.0,
     "name": "holiday-activated-48043",
     "write": true,
     "mappings": {
       "0": "inactive",
+      "1": "active",
       "10": "active"
```
Comment (yozik04) [inline nibe/data/f1155_f1255.json:5409]: We need to remove 10 then... I think

File: nibe/data/f750.json [hunk 2]
```diff
@@ -4447,16 +4447,17 @@
...
     "min": 0.0,
-    "max": 10.0,
+    "max": 1.0,
     "default": 0.0,
     "name": "holiday-activated-48043",
     "write": true,
     "mappings": {
       "0": "inactive",
+      "1": "active",
       "10": "active"
```
Comment (yozik04) [inline nibe/data/f750.json:4461]: Here as well.

File: nibe/data/f1155_f1255.json (hunk 1 above)
Comment (elupus) [reply nibe/data/f1155_f1255.json:5409]: Indeed, forgot

File: nibe/console_scripts/convert_csv.py [hunk 3]
```diff
@@ -16,7 +16,10 @@
 def update_dict(d: MutableMapping, u: Mapping, removeExplicitNulls: bool) -> Mapping:
     for k, v in u.items():
         if v is None and removeExplicitNulls:
-            d.pop(k, None)
+            try:
+                d.pop(k)
+            except (IndexError, KeyError):
+                pass
```
Comment (elupus) [inline nibe/console_scripts/convert_csv.py:22]: Turns out the d object is a pandas object here and not a dict. pop there does not take a default argument.  I think we don't recursively flatten the pandas object to a nested dict.

**Comment by ThunderMikey** [comment]
> 
What do you think, should we just add this to all F-series?

@elupus I'm happy to test the holiday extensions for F730. Any plan to add this to all F-series?

**Pull request merged**